src/compiler/checker.ts

```diff
@@ -19209,6 +19224,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         }
     }
 
+    function maybeCloneInferTypeParameter(p: TypeParameter) {
+        return getConstraintDeclaration(p) && couldContainTypeVariables(getConstraintFromTypeParameter(p)!) ? cloneTypeParameter(p) : p;
+    }

```

Comment:
this only clones type parameters with "local" constraints (the ones added by `infer R extends C` syntax)

It could do the same with all constraints, using this patch:

<details>
<summary>git diff</summary>

```diff
diff --git a/src/compiler/checker.ts b/src/compiler/checker.ts
index a8803f5823..52f44bcebe 100644
--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -19117,9 +19117,6 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     }
                 }
                 const context = createInferenceContext(freshParams, /*signature*/ undefined, InferenceFlags.None);
-                if (mapper) {
-                    context.nonFixingMapper = combineTypeMappers(context.nonFixingMapper, mapper);
-                }
                 if (!checkTypeDeferred) {
                     // We don't want inferences from constraints as they may cause us to eagerly resolve the
                     // conditional type instead of deferring resolution. Also, we always want strict function
@@ -19220,7 +19217,8 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function maybeCloneInferTypeParameter(p: TypeParameter) {
-        return getConstraintDeclaration(p) && couldContainTypeVariables(getConstraintFromTypeParameter(p)!) ? cloneTypeParameter(p) : p;
+        const constraint = getConstraintFromTypeParameter(p);
+        return constraint && couldContainTypeVariables(constraint) ? cloneTypeParameter(p) : p;
     }
 
     function getTrueTypeFromConditionalType(type: ConditionalType) {
diff --git a/tests/baselines/reference/inferTypes1.types b/tests/baselines/reference/inferTypes1.types
index edc97a2ea1..f6018d54a3 100644
--- a/tests/baselines/reference/inferTypes1.types
+++ b/tests/baselines/reference/inferTypes1.types
@@ -1,5 +1,8 @@
 //// [tests/cases/conformance/types/conditional/inferTypes1.ts] ////
 
+=== Performance Stats ===
+Instantiation count: 1,000
+
 === inferTypes1.ts ===
 type Unpacked<T> =
 >Unpacked : Unpacked<T>
```
</details>

As we can see, we could drop the `context.nonFixingMapper` manipulation and simplify the logic here - but that would create *more* type parameter clones and that's even caught by this diff (Instantiation count was reported by one of the tests when using this patch)